grafana-plugin/src/models/alert_receive_channel/alert_receive_channel.ts

```diff
@@ -105,23 +105,60 @@ export class AlertReceiveChannelStore {
     return alertReceiveChannel.data;
   }
 
-  async fetchServiceNowListOfStatus(): Promise<void> {
-    this.serviceNowStatusList = [
-      {
-        id: 1,
-        name: 'Resolved',
-      },
-      {
-        id: 2,
-        name: 'In Progress',
-      },
-      {
-        id: 3,
-        name: 'New',
-      },
-    ];
+  async fetchServiceNowStatusList({
+    id,
+    skipErrorHandling,
+  }: {
+    id: ApiSchemas['AlertReceiveChannel']['id'];
+    skipErrorHandling?: boolean;
+  }): Promise<void> {
+    const statusList = await onCallApi({ skipErrorHandling }).GET('/alert_receive_channels/{id}/status_options/', {
+      params: { path: { id } },
+    });
 
-    return Promise.resolve();
+    runInAction(() => {
+      // @ts-ignore // looks like wrong schema
+      this.serviceNowStatusList = statusList.data;
+    });
+  }
+
+  async hasServiceNowToken({ id }: { id: ApiSchemas['AlertReceiveChannel']['id'] }) {
+    try {
+      const response = await onCallApi({ skipErrorHandling: true }).GET('/alert_receive_channels/{id}/api_token/', {
+        params: { path: { id } },
+      });
+      return response?.response.status === 200;
+    } catch (ex) {
+      return false;
+    }
+  }
+
+  async testServiceNowAuthentication({ data }: { data: OmitReadonlyMembers<ApiSchemas['AlertReceiveChannelUpdate']> }) {
+    try {
+      const result = await onCallApi({ skipErrorHandling: false }).POST('/alert_receive_channels/test_connection/', {
+        body: data as ApiSchemas['AlertReceiveChannelUpdate'],
+        params: {},
+      });
+      return result?.response.status === 200;
+    } catch (ex) {
+      return false;
+    }
+  }
+
+  @AutoLoadingState(ActionKey.UPDATE_SERVICENOW_TOKEN)
+  @WithGlobalNotification({ failure: 'There was an error generating the token. Please try again' })
+  async generateServiceNowToken({
```

Comment:
it should either update some observable instead of returning data from this store or should be moved to alert_receive_channel.helpers.ts

Reply:
My bad, totally forgotten about this pattern we recently adopted. Done